src/distilabel/llm/anthropic.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2023-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import TYPE_CHECKING, List, Optional, Union
+
+from pydantic import PrivateAttr, SecretStr
+
+from distilabel.llm.base import AsyncLLM
+
+if TYPE_CHECKING:
+    from anthropic import AsyncAnthropic
+
+    from distilabel.llm.typing import GenerateOutput
+    from distilabel.steps.task.typing import ChatType
+
+
+class AnthropicLLM(AsyncLLM):
+    """Anthropic LLM implementation running the Async API client.
+
+    Args:
+        model: the model name to use for the LLM.
+        api_key: the API key to authenticate the requests to the Anthropic API.
+        base_url: the base URL to use for the Anthropic API. Defaults to "https://api.anthropic.com".
+        http_client: the HTTP client to use for the Anthropic API. Defaults to None.
+        timeout: the maximum time in seconds to wait for a response. Defaults to 600.0.
+        max_retries: the maximum number of retries for the LLM. Defaults to 2.
+    """
+
+    model: str = "claude-3-opus-20240229"
+    api_key: Optional[SecretStr] = os.getenv("ANTHROPIC_API_KEY", None)  # type: ignore
+    base_url: str = "https://api.anthropic.com"
+    timeout: float = 600.0
+    http_client: Union[str, None] = None
+    max_retries: int = 2
+
+    _aclient: Optional["AsyncAnthropic"] = PrivateAttr(...)
+
+    def load(self, api_key: Optional[str] = None) -> None:
+        """Loads the `AsyncAnthropic` client to use the Anthropic async API."""
+        try:
+            from anthropic import AsyncAnthropic
+        except ImportError as ie:
+            raise ImportError(
+                "Anthropic Python client is not installed. Please install it using"
+                " `pip install anthropic`."
+            ) from ie
+
+        self.api_key = self._handle_api_key_value(
+            self_value=self.api_key, load_value=api_key, env_var="ANTHROPIC_API_KEY"
```

Comment:
Could you set a `PrivateAttr` named `_env_var` with the `ANTHROPIC_API_KEY` as in https://github.com/argilla-io/distilabel/blob/1e52450400a0b324798268048ffa06da0297181a/src/distilabel/llm/openai.py#L43